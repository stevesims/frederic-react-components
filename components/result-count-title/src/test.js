import React from 'react';
import { mount, shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import ResultCountTitle from '.';

expect.extend(createMatchers(emotion));

describe('ResultCountTitle', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ResultCountTitle />);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders with count', () => {
    const wrapper = shallow(<ResultCountTitle count="000" />);
    expect(wrapper.find('ResultCount').childAt(0).text()).toBe('000');
  });

  it('renders with title', () => {
    const wrapper = shallow(<ResultCountTitle>Title</ResultCountTitle>);
    expect(wrapper.find('Title').childAt(0).text()).toBe('Title');
  });

  it('matches snapshot', () => {
    const wrapper = mount(<ResultCountTitle count="000">Title</ResultCountTitle>);
    expect(wrapper).toMatchSnapshot();
  });
});
