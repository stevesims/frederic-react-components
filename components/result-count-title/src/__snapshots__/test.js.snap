// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCountTitle matches snapshot 1`] = `
.emotion-4 {
  display: inline-block;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-0 {
  background: #6f777b;
  color: #ffffff;
  display: inline-block;
  font-family: "nta",Arial,sans-serif;
  font-size: 15px;
  line-height: 1.25;
  min-height: 20px;
  min-width: 20px;
  padding: 1px 4px;
  text-align: center;
}

.emotion-2 {
  display: inline-block;
  font-family: "nta",Arial,sans-serif;
  font-size: 15px;
  line-height: 1.25;
  margin-left: 10px;
}

<ResultCountTitle
  count="000"
>
  <Wrapper>
    <div
      className="emotion-4 emotion-5"
    >
      <ResultCount
        backgroundColor="#6f777b"
        color="#ffffff"
      >
        <StyledResultCount
          backgroundColor="#6f777b"
          color="#ffffff"
        >
          <span
            className="emotion-0 emotion-1"
            color="#ffffff"
          >
            000
          </span>
        </StyledResultCount>
      </ResultCount>
      <Title>
        <div
          className="emotion-2 emotion-3"
        >
          Title
        </div>
      </Title>
    </div>
  </Wrapper>
</ResultCountTitle>
`;
